main.go

```diff
@@ -266,3 +289,94 @@ func getTaskContributorsHandler(w http.ResponseWriter, r *http.Request, db *sql.
 		return
 	}
 }
+
+// URL Funcs
+
+func urlFormHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		tmpl := template.Must(template.ParseFiles("url.html"))
+		tmpl.Execute(w, nil)
+	} else {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+	}
+}
+
+func submitURLHandler(db *sql.DB) http.HandlerFunc {
```

Comment:
This method is too long.
It contains also some logic which needs to be extracted in a model or a service.